src/gui/editors/PianoRoll.cpp

```diff
@@ -868,6 +878,35 @@ void PianoRoll::drawNoteRect( QPainter & p, int x, int y,
 
 	p.drawRect( x, y, noteWidth, noteHeight );
 
+	// Draw note key text
+	if (drawNoteName)
+	{
+		p.save();
+		int const noteTextHeight = static_cast<int>(noteHeight * 0.8);
+		if (noteTextHeight > 6)
+		{
+			QString noteKeyString = getNoteString(n->key());
+
+			QFont noteFont(p.font());
+			noteFont.setPixelSize(noteTextHeight);
+			QFontMetrics fontMetrics(noteFont);
+			QSize textSize = fontMetrics.size(Qt::TextSingleLine, noteKeyString);
+
+			int const distanceToBorder = 2;
+			int const xOffset = borderWidth + distanceToBorder;
+
+			if (textSize.width() < noteWidth - xOffset)
+			{
+				p.setPen(noteTextColor);
+				p.setFont(noteFont);
+				QPoint textStart(x + xOffset, y + (noteTextHeight + (noteHeight - noteTextHeight) / 2));
```

Comment:
For what it's worth, the calculation of the y-coordinate (baseline) could be expressed more succinctly as `y + (noteHeight + noteTextHeight) / 2`.

Reply:
Done with 27bd8aa9505d6af8276b27c4d2472128c78ced1b.